### PR TITLE
v3.0 - Support PHP 8.1

### PR DESCRIPTION
**Pull request by jaxwilko**
Draft PR to include support for php8.1

### CODE REVIEW

**Comment by LukeTowers** [comment]
Still to do: evaluate and decide which CSS minifiers are worth of maintaining filters for. Also include for evaluation the custom one included with Winter CMS.

File: composer.json [hunk 1]
```diff
@@ -23,43 +23,42 @@
         }
     ],
     "require": {
-        "php": "^7.2 || ^8.0",
+        "php": ">=7.3 || ^8.0",
         "ext-curl": "*",
         "ext-json": "*",
         "ext-simplexml": "*",
-        "symfony/process": "~3.4 || ~4.0 || ~5.0",
+        "symfony/process": "~3.4 || ~4.0 || ~5.0 || ~6.0",
```
Comment (LukeTowers) [inline composer.json:30]: Should also update the minimum supported versions of these libraries based on our new minimum PHP requirement

File: src/Assetic/Extension/Twig/ValueContainer.php [hunk 2]
```diff
@@ -17,13 +18,15 @@ public function __construct(ValueSupplierInterface $valueSupplier)
...
 
-    public function offsetExists($offset)
+    public function offsetExists($offset): bool
     {
         $this->initialize();
 
         return array_key_exists($offset, $this->values);
     }
 
+    // Return type should change to :mixed as soon as PHP 8.0 is the lowest version targetted
```
Comment (bennothommo) [inline src/Assetic/Extension/Twig/ValueContainer.php]:
```suggestion
    // Return type should change to :mixed as soon as PHP 8.0 is the lowest version targeted
```

File: src/Assetic/Extension/Twig/ValueContainer.php [hunk 3]
```diff
@@ -35,24 +38,27 @@ public function offsetGet($offset)
...
     }
 
-    public function offsetSet($offset, $value)
+    public function offsetSet($offset, $value): void
     {
         throw new \BadMethodCallException('The ValueContainer is read-only.');
     }
 
+    // Return type should change to :void (and mixed $offset for parameter type
+    // hint) as soon as PHP 8.0 is the lowest version targetted
```
Comment (bennothommo) [inline src/Assetic/Extension/Twig/ValueContainer.php]:
```suggestion
    // hint) as soon as PHP 8.0 is the lowest version targeted
```

File: src/Assetic/Asset/Iterator/AssetCollectionIterator.php [hunk 4]
```diff
@@ -70,35 +73,37 @@ public function current($raw = false)
         return $clone;
     }
 
+    // Return type should change to :mixed as soon as PHP 8.0 is the lowest version targetted
```
Comment (bennothommo) [inline src/Assetic/Asset/Iterator/AssetCollectionIterator.php]:
```suggestion
    // Return type should change to :mixed as soon as PHP 8.0 is the lowest version targeted
```

File: src/Assetic/Asset/Iterator/AssetCollectionIterator.php [hunk 5]
```diff
@@ -40,6 +41,8 @@ public function __construct(AssetCollectionInterface $coll, \SplObjectStorage $c
      *
      * @return \Assetic\Asset\AssetInterface
      */
+    // Return type should change to :mixed as soon as PHP 8.0 is the lowest version targetted
```
Comment (bennothommo) [inline src/Assetic/Asset/Iterator/AssetCollectionIterator.php]:
```suggestion
    // Return type should change to :mixed as soon as PHP 8.0 is the lowest version targeted
```

File: src/Assetic/Factory/Resource/DirectoryResourceIterator.php [hunk 6]
```diff
@@ -8,6 +8,8 @@
  */
 class DirectoryResourceIterator extends \RecursiveIteratorIterator
 {
+    // Return type should change to :mixed as soon as PHP 8.0 is the lowest version targetted
```
Comment (bennothommo) [inline src/Assetic/Factory/Resource/DirectoryResourceIterator.php]:
```suggestion
    // Return type should change to :mixed as soon as PHP 8.0 is the lowest version targeted
```

**Pull request merged**